src/docfx/restore/RestoreMap.cs

```diff
@@ -9,40 +9,47 @@
 
 namespace Microsoft.Docs.Build
 {
-    internal class RestoreMap
+    internal static class RestoreMap
     {
-        private static readonly ConcurrentDictionary<string, Lazy<string>> s_gitPath = new ConcurrentDictionary<string, Lazy<string>>();
+        private static readonly ConcurrentDictionary<(string remote, string branch), Lazy<string>> s_gitPath = new ConcurrentDictionary<(string remote, string branch), Lazy<string>>();
         private static readonly ConcurrentDictionary<string, Lazy<string>> s_downloadPath = new ConcurrentDictionary<string, Lazy<string>>();
 
-        private readonly string _docsetPath;
-
-        public RestoreMap(string docsetPath)
+        public static string GetGitRestorePath(string url)
         {
-            Debug.Assert(!string.IsNullOrEmpty(docsetPath));
-            _docsetPath = docsetPath;
+            var (remote, branch) = GitUtility.GetGitRemoteInfo(url);
+            return GetGitRestorePath(remote, branch);
         }
 
-        public string GetGitRestorePath(string url)
+        public static string GetGitRestorePath(string remote, string branch)
         {
-            Debug.Assert(!string.IsNullOrEmpty(url));
+            if (!TryGetGitRestorePath(remote, branch, out var result))
+            {
+                throw Errors.NeedRestore($"{remote}#{branch}").ToException();
+            }
+            return result;
+        }
 
-            var gitRestorePath = s_gitPath.GetOrAdd(url, new Lazy<string>(FindLastModifiedGitRepository)).Value;
+        public static bool TryGetGitRestorePath(string remote, string branch, out string result)
+        {
+            result = s_gitPath.GetOrAdd((remote, branch), new Lazy<string>(FindLastModifiedGitRepository)).Value;
```

Comment:
>GetOrAdd [](start = 31, length = 8)

use AddOrUpdate

Reply:
It does not solve the null cache problem

Reply:
if it's null, you can use update function to restore again